Signal/src/ViewControllers/OWSConversationSettingsViewController.m

```diff
@@ -1124,6 +1133,19 @@ - (void)identityStateDidChange:(NSNotification *)notification
     [self updateTableContents];
 }
 
+- (void)otherUsersProfileDidChange:(NSNotification *)notification
+{
+    OWSAssert([NSThread isMainThread]);
+
+    NSString *recipientId = notification.userInfo[kNSNotificationKey_ProfileRecipientId];
+    OWSAssert(recipientId.length > 0);
+
+    if (recipientId.length > 0 && [self.thread isKindOfClass:[TSContactThread class]] &&
+        [self.thread.contactIdentifier isEqualToString:recipientId]) {
```

Comment:
If we're going to do this a lot, we could create a category on `NSNotification+Profiles` with a `appliesToGroup:`, `appliesToUser:` and/or `appliesToThread:` methods.